### ISSUE TITLE
Erroneous errors with clang-cl.exe

### ISSUE THREAD

**Issue by augusew1** [pboettch/json-schema-validator#378]
[This line in `src/CMakeLists.txt`](https://github.com/pboettch/json-schema-validator/blob/main/src/CMakeLists.txt#L28) is incorrect when using clang-cl.exe on Windows. 

It results in a tremendous amount of warnings about c++98-compat and a bunch of other bogus things because it is passing a unix-like flag to an MSVC-like front-end.

Internally, we are using this work around:

```cmake
    if (("${CMAKE_CXX_COMPILER_FRONTEND_VARIANT}" STREQUAL "MSVC") AND ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
        # Bug in the library where they treat Clang as if it isn't on Windows unconditionally, we have to turn off
        # all of the warnings. Not ideal.
        # TODO: report this upstream
        target_compile_options(nlohmann_json_schema_validator PRIVATE "-w" "/w")
    endif()
```
The proper fix would be to check for[ `CMAKE_CXX_COMPILER_FRONTEND_VARIANT`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_FRONTEND_VARIANT.html), which, thankfully, is in CMake 3.14. No bump of CMake is required to support this.
